src/linear_samplers/block_col_sample_wo_replacement.jl

```diff
@@ -0,0 +1,107 @@
+# This file is part of RLinearAlgebra.jl
+# 1. Specifies type
+# 2. Implements sample function
+# 3. Exports type
+#
+# Date: 07/14/2024
+# Author: Christian Varner
+# Purpose: Implement a col sketching algorithm that selects
+# columns without replacement from an arbitrary probability distribution.
+
+"""
+    LinSysBlkColSelectWoReplacement <: LinSysVecColSelect
+
+A mutable struct that represents sampling columns from `A` without replacement using an
+arbitrary weight/probability vector.
+
+# Fields
+
+- `block_size::Int64`, number of columns sampled (i.e., number of columns in `A * S`)
+- `probability::Union{Weights, Vector{Float64}, Nothing}`, vector that represents a probability distribution over the columns of `A`. Requirements are that the probabilities sum to 1, are non-negative, `probability` has the same length as number of columns in `A`, and `probability` has at least as many positive entries as `block_size`. If `probability` is unspecified in the constructor, `sample` will default to a uniform distribution over columns of `A`.
```

Comment:
Do you mean that by default, this will select uniformly from the columns of `A` without replacement?

Reply:
Yes